### PR TITLE
[IOTDB-4073] Add SessionTimeWindowAccessStrategy in UDF

### PR DESCRIPTION
**Pull request by ZhanGHanG9991**
## Description
Add SessionTimeWindowAccessStrategy in UDF

<hr>

This PR has:
- [x] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [x] added or updated version, __license__, or notice information
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [x] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnMultiReferenceIntermediateLayer.java [hunk 1]
```diff
@@ -542,4 +543,120 @@ public RowWindow currentWindow() {
...
+          nextWindowTimeBegin = Math.max(displayWindowBegin, tvList.getTime(0));
+          hasAtLeastOneRow = tvList.size() != 0;
+          isFirstIteration = false;
+        }
+
+        if (!hasAtLeastOneRow || displayWindowEnd <= nextWindowTimeBegin) {
+          return YieldableState.NOT_YIELDABLE_NO_MORE_DATA;
+        }
+
+        while (tvList.getTime(tvList.size() - 1) < displayWindowEnd) {
```
Comment (lancelly) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnMultiReferenceIntermediateLayer.java:588]: If using displayWindowEnd here, all data will be stored first

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java [hunk 2]
```diff
@@ -653,4 +654,122 @@ public RowWindow currentWindow() {
...
+      @Override
+      public TSDataType[] getDataTypes() {
+        return new TSDataType[0];
+      }
+
+      @Override
+      public RowWindow currentWindow() {
+        return null;
+      }
+    };
```
Comment (lancelly) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java:773]:
```suggestion
      @Override
      public RowWindow currentWindow() {
        return window;
      }
    };
```

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java [hunk 3]
```diff
@@ -653,4 +654,122 @@ public RowWindow currentWindow() {
...
+          nextWindowTimeBegin = rowRecordList.getTime(nextIndexEnd);
+        }
+        rowRecordList.setEvictionUpperBound(nextIndexBegin + 1);
+        nextIndexBegin = nextIndexEnd;
+      }
+
+      @Override
+      public TSDataType[] getDataTypes() {
+        return new TSDataType[0];
+      }
```
Comment (lancelly) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java:767]: why not return dataTypes?

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(IoTDBTestRunner.class)
+@Category({ClusterIT.class})
+public class IoTDBUDFSessionWindowQueryIT {
+
+  protected static final int ITERATION_TIMES = 10000;
```
Comment (lancelly) [inline integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:48]: ITERATION_TIMES could be smaller.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+    try (Connection conn = EnvFactory.getEnv().getConnection();
+        Statement statement = conn.createStatement();
+        ResultSet resultSet = statement.executeQuery(sql)) {
+      assertEquals(2, resultSet.getMetaData().getColumnCount());
+      for (int i = 0; i < windowStart.length; i++) {
+        resultSet.next();
+        Assert.assertEquals(resultSet.getLong(1), windowStart[i]);
+        Assert.assertEquals(resultSet.getLong(2), windowEnd[i]);
+      }
```
Comment (lancelly) [inline integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:153]: use while(resultSet.next()) and assert cnt instead.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+          || i == 9997 || i == 9998) {
+        continue;
+      }
+      windowStart.add(i);
+      windowEnd.add(i);
+    }
+    testSessionTimeWindowSS(
+        sessionGap,
+        windowStart.stream().mapToLong(t -> t.longValue()).toArray(),
+        windowEnd.stream().mapToLong(t -> t.longValue()).toArray(),
```
Comment (lancelly) [inline integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:222]:
```suggestion
        windowStart.stream().mapToLong(t -> t).toArray(),
        windowEnd.stream().mapToLong(t -> t).toArray(),
```

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+    } else {
+      sql =
+          String.format(
+              "select window_start_end(s3, '%s'='%s', '%s'='%s', '%s'='%s', '%s'='%s') from root.vehicle.d1",
+              UDFTestConstant.ACCESS_STRATEGY_KEY,
+              UDFTestConstant.ACCESS_STRATEGY_SESSION,
+              UDFTestConstant.DISPLAY_WINDOW_BEGIN_KEY,
+              displayBegin.longValue(),
+              UDFTestConstant.DISPLAY_WINDOW_END_KEY,
+              displayEnd.longValue(),
```
Comment (lancelly) [inline integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:247]:
```suggestion
              displayBegin,
              UDFTestConstant.DISPLAY_WINDOW_END_KEY,
              displayEnd,
```

**Review by lancelly — CHANGES_REQUESTED** [review]
PTAL~

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnMultiReferenceIntermediateLayer.java (hunk 1 above)
Comment (ZhanGHanG9991) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnMultiReferenceIntermediateLayer.java:588]: The if statement below while loop has break. We needn't store all data.

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java (hunk 2 above)
Comment (ZhanGHanG9991) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java:773]: I've corrected it. Thanks.

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java (hunk 3 above)
Comment (ZhanGHanG9991) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/MultiInputColumnIntermediateLayer.java:767]: I've corrected it. Thanks.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java (hunk 4 above)
Comment (ZhanGHanG9991) [reply integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:48]: I've made it take 1000.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java (hunk 5 above)
Comment (ZhanGHanG9991) [reply integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java:153]: I've changed it.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java (hunk 6 above)
Comment (ZhanGHanG9991) [reply integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java]: I've changed it.

File: integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java (hunk 7 above)
Comment (ZhanGHanG9991) [reply integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java]: I've changed.

**Review by lancelly — APPROVED** [review]
LGTM~

**Pull request merged**